examples/multi_gpu/papers100m_gcn.py

```diff
@@ -1,142 +1,331 @@
+                          ", Loss: " + str(loss))
+        dist.barrier()
+        with Join([model]):
+            if rank == 0:
+                print("Average Training Iteration Time:",
+                      (time.time() - start) / (i - warmup_steps), "s/iter")
+                acc_sum = 0.0
                 with torch.no_grad():
-                    out = model(batch.x, batch.edge_index)[:batch.batch_size]
-                pred = out.argmax(dim=-1)
-                y = batch.y[:batch.batch_size].view(-1).to(torch.long)
+                    for i, batch in enumerate(eval_loader):
+                        if i >= eval_steps:
+                            break
 
-                total_correct += int((pred == y).sum())
-                total_examples += y.size(0)
+                        batch = batch.to(rank)
+                        if isinstance(batch, torch_geometric.data.HeteroData):
+                            batch = batch.to_homogeneous()
+                        batch_size = batch.num_sampled_nodes[0]
 
-            print(f"Val Acc: {total_correct / total_examples:.4f}")
-            sec_per_iter = (time.time() - start) / (i - warmup_steps)
-            print(f"Avg Inference Iteration Time: {sec_per_iter:.6f} s/iter")
+                        batch.y = batch.y.to(torch.long)
+                        out = model.module(batch.x, batch.edge_index)
+                        acc_sum += acc(out[:batch_size].softmax(dim=-1),
+                                       batch.y[:batch_size])
+                print(f"Validation Accuracy: {acc_sum/(i) * 100.0:.4f}%", )
+        dist.barrier()
 
-    if rank == 0:
-        model.eval()
-        total_correct = total_examples = 0
-        for i, batch in enumerate(test_loader):
-            batch = batch.to(rank)
+    with Join([model]):
+        if rank == 0:
+            acc_sum = 0.0
             with torch.no_grad():
-                out = model(batch.x, batch.edge_index)[:batch.batch_size]
-            pred = out.argmax(dim=-1)
-            y = batch.y[:batch.batch_size].view(-1).to(torch.long)
+                for i, batch in enumerate(test_loader):
+                    batch = batch.to(rank)
+                    if isinstance(batch, torch_geometric.data.HeteroData):
+                        batch = batch.to_homogeneous()
+                    batch_size = batch.num_sampled_nodes[0]
 
-            total_correct += int((pred == y).sum())
-            total_examples += y.size(0)
-        print(f"Test Acc: {total_correct / total_examples:.4f}")
+                    batch.y = batch.y.to(torch.long)
+                    out = model.module(batch.x, batch.edge_index)
+                    acc_sum += acc(out[:batch_size].softmax(dim=-1),
+                                   batch.y[:batch_size])
+                print(f"Test Accuracy: {acc_sum/(i) * 100.0:.4f}%", )
+    if rank == 0:
+        print("Total Program Runtime =",
+              round(time.perf_counter() - wall_clock_start, 2), "seconds")
```

Comment:
small comment: As your current script starts measuring time before launching processes, it should also measure time until after teardown.